modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/IgnitePdsBinaryMetadataOnClusterRestartTest.java

```diff
@@ -334,24 +336,31 @@ public void testNodeWithIncompatibleMetadataIsProhibitedToJoinTheCluster() throw
 
         startGridInASeparateWorkDir("A");
 
-        boolean exceptedExceptionThrown = false;
+        String expectedMsg = String.format(
+            "Type '%s' with typeId %d has a different/incorrect type for field '%s'. Expected 'int' but 'long' was " +
+                "provided. Field type's modification is unsupported, clean {root_path}/marshaller and " +
+                "{root_path}/binary_meta directories if the type change is required.",
+            DYNAMIC_TYPE_NAME,
+            createdTypeId,
+            decimalFieldName);
+
+        boolean expectedExcThrown = false;
+
         try {
             startGridInASeparateWorkDir("B");
         }
         catch (Exception e) {
```

Comment:
Ignite has method GridTestUtils.assertThrows() to check if correct exception has been thrown.

Reply:
Updated according to your comments.